OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/FirebaseAnalytics.java

```diff
@@ -0,0 +1,378 @@
+     * @param courseId   course id of the course we are viewing
+     * @param action     any custom action we need to send with event
+     * @param values     any custom key-value pairs we need to send with event
+     */
+    @Override
+    public void trackScreenView(@NonNull String screenName, @Nullable String courseId,
+                                @Nullable String action,
+                                @Nullable Map<String, String> values) {
+        final FirebaseEvent event = new FirebaseEvent(screenName);
+        if (!TextUtils.isEmpty(action)) {
+            event.putString(Keys.ACTION, action);
+        }
+        if (!TextUtils.isEmpty(courseId)) {
+            event.putCourseId(courseId);
+        }
+        if (values != null) {
+            event.putMap(values);
+        }
+
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoLoading(String videoId, String courseId, String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.LOADED_VIDEO, videoId, Values.VIDEO_LOADED);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoPlaying(String videoId, Double currentTime,
+                                  String courseId, String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.PLAYED_VIDEO, videoId,
+                Values.VIDEO_PLAYED, currentTime);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoPause(String videoId,
+                                Double currentTime, String courseId, String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.PAUSED_VIDEO,
+                videoId, Values.VIDEO_PAUSED, currentTime);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoStop(String videoId, Double currentTime, String courseId,
+                               String unitUrl) {
+        final FirebaseEvent event = new FirebaseEvent(Events.STOPPED_VIDEO,
+                videoId, Values.VIDEO_STOPPED, currentTime);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        tracker.logEvent(event.getName(), event.getBundle());
+    }
+
+    @Override
+    public void trackVideoSeek(String videoId, Double oldTime, Double newTime, String courseId,
+                               String unitUrl, Boolean skipSeek) {
+        final FirebaseEvent event = new FirebaseEvent(Events.SEEK_VIDEO, videoId, Values.VIDEO_SEEKED);
+        event.setCourseContext(courseId, unitUrl, Values.VIDEOPLAYER);
+        //Call the format Double value so that we can have upto 3 decimal places after
```

Comment:
Are you talking about the comment or the function?
Cuz the function `formatDoubleValue` was moved into `JavaUtil` function in a later commit i guess.